pkg/engine/mutate/mutation.go

```diff
@@ -94,13 +94,11 @@ func ForEach(name string, foreach kyvernov1.ForEachMutation, policyContext engin
 	if strings.TrimSpace(string(resourceBytes)) == strings.TrimSpace(string(patchedBytes)) {
 		return NewResponse(engineapi.RuleStatusSkip, resource, "no patches applied")
 	}
-	if err := resource.UnmarshalJSON(patchedBytes); err != nil {
+	if err := patchedResource.UnmarshalJSON(patchedBytes); err != nil {
 		return NewErrorResponse("failed to unmarshal patched resource", err)
-	} else if err := ctx.AddResource(resource.Object); err != nil {
-		return NewErrorResponse("failed to update patched resource in the JSON context", err)
```

Comment:
Missing adding the resource context?

Reply:
No, the context resource is immutable. We should not be updating it directly.